iOS/ZappAnalyticsPluginFirebase/APAnalyticsProviderFirebase.swift

```diff
@@ -131,6 +141,14 @@ open class APAnalyticsProviderFirebase: ZPAnalyticsProvider {
                 }
                 Analytics.setUserProperty(value, forName: key)
             }
+            
+            //We want to add the user id only if it is configured as allow-tracking-user-id-for-app-family-<APP_FAMILY_ID> in the plugin configurations
+            if let userIDParameter = configurationJSON?[UserIDConstants.userIDJSONConfig] as? String,
```

Comment:
Put this logic in a fileprivate method and call this here.
Add error logs.
Use guard statement and separate this logic.

Reply:
To abstract the logic to a method is fine, but I don't understand what you want to check in an error log there. I don't think we should track any error if this logic is not executing.

Reply:
Commited a new change with refactor of the method.